dbt/adapters/trino/connections.py

```diff
@@ -460,11 +459,17 @@ def add_query(self, sql, auto_begin=True, bindings=None, abridge_sql_log=False):
             )
 
         if cursor is None:
-            raise dbt.exceptions.RuntimeException(
+            conn = self.get_thread_connection()
```

Comment:
Is the exception handling related to dbt 1.3 upgrade?
If not, let's do it separately to leave more time to discuss it in detail.

Reply:
This is something we already discussed before, @hovaesco  had this issue and it seems resolved exactly like this in the dbt's snowflake adapter. There are probably some conditions within dbt that may result in having no thread connection. It is unclear how to reproduce it but probably something really happening in the field.